src/util/getUrl.ts

```diff
@@ -13,10 +16,27 @@ export const getGitHubConfigurationUrl = (urlParams: UrlParams): string => {
 export const getJiraMarketplaceUrl = (jiraHost: string): string =>
 	`${jiraHost}/plugins/servlet/ac/com.atlassian.jira.emcee/discover#!/discover/app/com.github.integration.production`;
 
+// Deprecated, will be removed with the feature flag. Use getJiraHostFromRedirectUrlNew() instead
 export const getJiraHostFromRedirectUrl = (url: string): string => {
 	try {
 		return new URL(url).host;
 	} catch (err) {
 		return "unknown";
 	}
 };
+
+export const getJiraHostFromRedirectUrlNew = (urlOrPath: string, log: Logger): string => {
+	if (!urlOrPath) {
+		return "empty";
+	}
+	try {
+		const { host, query } = url.parse(urlOrPath, true);
+		if (query && query.xdm_e) {
+			return "" + url.parse("" + query.xdm_e, false).host;
```

Comment:
isn't host already a string?  then again, it would be optional right?

Can't the whole algorithm be just simplified to 
```
const { host, query } = url.parse(urlOrPath, true);
return (query?.xdm_e ? url.parse(query.xdm_e.toString(), false).host : host) || "unknown";
```

Reply:
How come GitHub doesn't have :awesome: icon?